core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/AbstractSampleToSplitPointsHandler.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017-2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.gaffer.store.operation.handler;
+
+import uk.gov.gchq.gaffer.commonutil.stream.Streams;
+import uk.gov.gchq.gaffer.operation.OperationException;
+import uk.gov.gchq.gaffer.operation.impl.SampleToSplitPoints;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.Store;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public abstract class AbstractSampleToSplitPointsHandler <T, S extends Store> implements OutputOperationHandler<SampleToSplitPoints<T>, List<T>> {
+
+    @Override
+    public List<T> doOperation(final SampleToSplitPoints<T> operation, final Context context, final Store store) throws OperationException {
+
+        final S typedStore = (S) store;
+
+        validate(operation, typedStore);
+
+        final Integer numSplits = getNumSplits(operation, typedStore);
+        if (null == numSplits) {
+            throw new OperationException("Number of splits is required");
+        }
+        if (numSplits < 1) {
+            return Collections.emptyList();
+        }
+
+        final List<T> records = Streams.toStream(operation.getInput()).collect(Collectors.toList());
```

Comment:
How big is input expected to be, pulling the iterator into memory.

Reply:
The code in AbstractSampleToSplitPointsHandler was refactored out of AbstractSampleElementsToSplitPointsHandler which already imposed a maximum sample limit before executing this code.  
For the two new operations SplitStoreFromRDDOfElementsHandler and SplitStoreFromJavaRDDElementsHandler these also impose maximum sample limits before calling the operation.  
It's possible a user might generate their own large sample, use this operation directly and blow the memory. Is that something we should guard against?

Reply:
My comment was more of a sense check and highlight for the reviewer.

Reply:
@m29827 Yes if it's possible for a user to run this operation through the REST service, then it should be guarded against. However this can be disabled using a disableOperations.json file. There's one of these in the accumulo-rest, core-rest and hbase-rest modules (under rest-api). If you add the new operations to this list, then users will not be able to run them via the REST API.

Reply:
@d47853 added the operation to the disabled operation lists as suggested